src/task/models/tileMergeTaskManager.ts

```diff
@@ -243,14 +270,41 @@ export class TileMergeTaskManager {
 
     span.setAttributes({ partsZoomLevelMatch, maxZoom, ppCollectionLength: ppCollection.features.length });
 
-    for (let zoom = maxZoom; zoom >= 0; zoom--) {
+    // Store original resume state (NEVER changes during loop)
+    const resumeFromZoom = initTask.parameters.taskIndex?.zoomLevel ?? maxZoom;
+    const resumeFromTaskIndex = initTask.parameters.taskIndex?.lastInsertedTaskIndex ?? 0;
+
+    logger.info({
+      msg: 'Resume state initialized',
+      resumeFromZoom,
+      resumeFromTaskIndex,
+      maxZoom,
+    });
+
+    let zoom: number = resumeFromZoom;
+
+    for (zoom; zoom >= 0; zoom--) {
       logger.info({ msg: 'Processing zoom level', zoom });
+
       if (!partsZoomLevelMatch) {
         const filteredFeatures = ppCollection.features.filter((feature) => feature.properties.maxZoom >= zoom);
         const collection = featureCollection(filteredFeatures);
         unifiedPart = this.unifyParts(collection, tilesSource);
       }
-      yield* this.createTasksForPart(unifiedPart!, zoom, taskMetadata, span);
+
+      // Only skip tasks on the EXACT original resume zoom level
+      const shouldSkipTasks = zoom === resumeFromZoom;
```

Comment:
the fact zoom and resumeFromZoom are matched isn't a good indication that we need skip it

Reply:
we need to talk about it